### PR TITLE
VulkanDriver: remove all CPU work queues.

### PR DESCRIPTION
**Pull request by prideout**

### CODE REVIEW

File: filament/src/driver/vulkan/VulkanStagePool.cpp [hunk 1]
```diff
@@ -51,15 +51,22 @@ VulkanStage const* VulkanStagePool::acquireStage(uint32_t numBytes) noexcept {
...
-    auto iter = mUsedStages.find(stage);
-    if (iter == mUsedStages.end()) {
-        utils::slog.e << "Unknown stage: " << stage->capacity << " bytes" << utils::io::endl;
-        return;
-    }
-    stage->lastAccessed = mCurrentFrame;
-    mUsedStages.erase(iter);
-    mFreeStages.insert(std::make_pair(stage->capacity, stage));
+void VulkanStagePool::releaseStage(VulkanStage const* stage, VulkanCommandBuffer& cmd) noexcept {
+    mDisposer.createDisposable(stage, [stage, this]() {
```
Comment (bejado) [inline filament/src/driver/vulkan/VulkanStagePool.cpp]: Why isn't this disposable stage created in `acquireStage`?

Comment (prideout) [reply filament/src/driver/vulkan/VulkanStagePool.cpp]: At the time that we acquire a staging area, we don't know yet when we'll be done with it.

At the time that we release a staging area, we know that we can destroy it as soon as the current command buffer has finished executing.

Comment (prideout) [reply filament/src/driver/vulkan/VulkanStagePool.cpp]: Oh I see what you mean, yeah I think I could've structured it that way.

Comment (bejado) [reply filament/src/driver/vulkan/VulkanStagePool.cpp]: Gotcha, makes sense

**Pull request merged**